### PR TITLE
Added 3 new tools

### PR DESCRIPTION
**Pull request by palexiou**
Create command to join multiple tables
Create command to collapse identical fastq sequences
Create command to trim fastq sequences

### CODE REVIEW

File: lib/CLIPSeqTools/PreprocessApp/collapse_fastq.pm [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	if ($k == 0){
+		push @kmers, '.';
+	}
+	else {
+		@kmers = _permuteK($k);
+	}
+
+	my $count = 0;
+	foreach my $kmer (@kmers){
+		if ($kmer =~ /\s/){next;}
```
Comment (mnsmar) [inline lib/CLIPSeqTools/PreprocessApp/collapse_fastq.pm]: What is this check for?

File: lib/CLIPSeqTools/PreprocessApp/collapse_fastq.pm [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+		my $class = "GenOO::Data::File::FASTQ";
```
Comment (mnsmar) [inline lib/CLIPSeqTools/PreprocessApp/collapse_fastq.pm]: Why do you declare a variable named class? I don't see it being used elsewhere except line 109.

File: lib/CLIPSeqTools/PreprocessApp/collapse_fastq.pm [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+		my $fp = $class->new(file => $fastq);
+		my %already_found = ();
+		while (my $rec = $fp->next_record) {
+			my $seq = $rec->sequence;
+			if ($seq =~ /![ATGC]/){
```
Comment (mnsmar) [inline lib/CLIPSeqTools/PreprocessApp/collapse_fastq.pm:111]: What is this if statement doing?

Comment (palexiou) [inline lib/CLIPSeqTools/PreprocessApp/collapse_fastq.pm:111]: Sometimes fastq files contain Ns.

File: lib/CLIPSeqTools/PreprocessApp/collapse_fastq.pm (hunk 2 above)
Comment (palexiou) [inline lib/CLIPSeqTools/PreprocessApp/collapse_fastq.pm]: We can combine these two lines. It was copy-paste from some code that had possibility for multiple classes.

File: lib/CLIPSeqTools/PreprocessApp/collapse_fastq.pm (hunk 1 above)
Comment (palexiou) [inline lib/CLIPSeqTools/PreprocessApp/collapse_fastq.pm]: This remained from debugging. We can take it out - it will never go in there.

**Pull request merged**